src/Validator/UniqueUrlValidator.php

```diff
@@ -53,7 +53,12 @@ public function validate(mixed $value, Constraint $constraint): void
             return;
         }
 
-        $this->manager->fixUrl($value);
+        /**
+         * TODO: check if it is really necessary.
+         *  Because it is a validator, it shouldn't change the object.
+         *  and as it isn't persisting the changes on DB, it is changing the object before pass on the lines bellow.
+         */
+        // $this->manager->fixUrl($value);
```

Comment:
it was added along time ago

https://github.com/sonata-project/SonataPageBundle/commit/50dcc77ff0697cc4f5d81d2ec3e5597c339c981b#diff-a2851bfaa1a6b101fb0fd268d31da8db71b64a9765532be31d255da7c53a796bR50